src/tests/python/firewalld_direct.py

```diff
@@ -21,12 +21,12 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-# To use in git tree: PYTHONPATH=.. python firewalld-test.py
-
 import dbus
 import sys
 import unittest
 
+import fwtst  # noqa: F401
```

Comment:
`fwtst` is a module with helper/utils for tests. Files like `firewalld_direct.py` may not be "unit-tests", but they are still tests.

I think all tests should share code and use helpers from the test-helper-module (`fwtst`, that is). Writing a test that doesn't use helpers likely is doing something wrong. The `import` is there, to make it clear that this is a test and that it would be appropriate to use the `fwtst` helper module.

That does not only apply to unit tests (named `test_*.py`) that are automatically run `pytest`.

---

Also, the import of `fwtst` already does something useful (the F401 warning is wrong, because the mere import of the module adjusts the PYTHONPATH). I don't understand how https://docs.pytest.org/en/7.2.x/explanation/pythonpath.html helps with that, it seems to talk about a command line switch?

The point of the automatic adjustment of the python path is that its... automatic. It just works and does the right thing. If you run `FWTST_AUTO_PYTHONPATH=0 pytest` you'll see that it doesn't work without that automatism.

The entire thing in `fwtst.py` is a few, clear(?) lines of code. How else could this be achieved?

I don't want to write a shell script that sets me PYTHONPATH and only run unit tests via that script. Plain `pytest` should just work. Also, you can opt-out from that (`FWTST_AUTO_PYTHONPATH=0`), if for some reason want to import the firewalld module from /usr/lib (or set PYHONPATH= yourself).

Reply:
when those tests run, they don't use the sources from the source tree. Unless, the user goes out of its way, and sets `PYTHONPATH=`. There is even a code comment about that:

```
# To use in git tree: PYTHONPATH=.. python firewalld-test.py
```

this is something that should just work out-of-the-box. 

That a python program sets up `sys.path` so that it's module can be found, seems a very common thing. For example, my `/usr/bin/dnf` has
```python
here = sys.path[0]
if here == '/usr/bin':
    # we never import Python modules from /usr/bin
    # removing this lowers the risk of accidental imports of weird files
    del sys.path[0]
else:
    # git checkout
    import os
    dnf_toplevel = os.path.dirname(here)
    sys.path[0] = dnf_toplevel
```


> Nothing is being used from the imported module.

Importing already does something, and it does so for convenience. The alternative would be
```python
import fwtst

fwtst.setup_python_path()

import firewall...
```

then the module would be used, but I don't think that's better.